vendor/ocaml/typing/typecore.ml

```diff
@@ -3976,6 +3976,10 @@ let report_error env ppf = function
       fprintf ppf
         "@[Exception patterns must be at the top level of a match case.@]"
 
+#if undefined BS_NO_COMPILER_PATCH then
+let super_report_error_no_wrap_printing_env = report_error
+#end
+
 let report_error env ppf err =
   wrap_printing_env env (fun () -> report_error env ppf err)
```

Comment:
- let's split the changes to compiler in a separate commit
- we don't need such macros, a clear delimiter would be good
```
#if true then
#end
```
- why do you need do the renaming, can you just expose report_error

Reply:
There's already an exposed `report_error` right below. I'm exposing the non-wrapped version of report_error before it gets redefined

Reply:
The compiler change is at https://github.com/BuckleScript/ocaml/pull/22. You said it's automated? Can I just submit that to this repo then?

Reply:
https://github.com/BuckleScript/ocaml/commit/960bf54d1116b5d842c912d7acf12ac060fc09f8
Maybe you can do a cherry-pick to 4.06